program/src/instruction.rs

```diff
@@ -1,184 +1,373 @@
-//! Program instructions
+use borsh::{BorshDeserialize, BorshSerialize};
+use shank::{ShankContext, ShankInstruction};
 
-use solana_program::clock::UnixTimestamp;
-
-/// Instructions supported by the staking program
-#[derive(Clone, Debug, PartialEq, Eq)]
-pub enum StakeInstruction {
-    /// Creates Stake config account which controls staking parameters
-    ///
-    /// 0. `[w]` Stake config account
-    /// 1. `[]` Slash authority, may be a PDA for a governance program
-    /// 2. `[]` Config authority, may be a PDA for a governance program
-    /// 3. `[]` Stake Token Mint
-    /// 4. `[]` Stake Token Vault, to hold all staked tokens.
-    ///   Must be empty and owned by PDA with seeds `['token-owner', stake_config]`
+/// Enum defining all instructions in the Stake program.
+#[rustfmt::skip]
+#[derive(BorshDeserialize, BorshSerialize, Clone, Debug, ShankContext, ShankInstruction)]
+pub enum Instruction {
+    /// Creates Stake config account which controls staking parameters.
+    #[account(
+        0,
+        writable,
+        name = "config",
+        desc = "Stake config account"
+    )]
+    #[account(
+        1,
+        name = "config_authority",
+        desc = "Config authority"
+    )]
+    #[account(
+        2,
+        name = "slash_authority",
+        desc = "Slash authority"
```

Comment:
The order does not matter if you are using the `Context` derive, since you access accounts by name. I inverted them since so the `config_authority` comes right after  the `config` account, but this is totally cosmetic – I can switch it back if you prefer.

Reply:
Reverted back to the original order.